Jenkinsfile

```diff
@@ -24,23 +24,51 @@ stage('Build') {
 }
 
 stage('QA') {
-    node {
-        unstash name: 'built'
-        // findBugs
-        try {
-            sh './gradlew -Dfindbugs.xml.report=true findbugsMain'
-        } finally {
-            step([$class: 'FindBugsPublisher', pattern: '**/build/reports/findbugs/*.xml'])
-        }
-        // tests
-        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'clientlibs-test', usernameVariable: 'DB_USER', passwordVariable: 'DB_PASSWORD']]) {
-            try {
-                sh './gradlew -Dtest.with.specified.couch=true -Dtest.couch.username=$DB_USER -Dtest.couch.password=$DB_PASSWORD -Dtest.couch.host=clientlibs-test.cloudant.com -Dtest.couch.port=443 -Dtest.couch.http=https -Dtest.couch.ignore.compaction=true -Dtest.couch.auth.headers=true cloudantServiceTest'
-            } finally {
-                junit '**/build/test-results/*.xml'
-            }
-        }
-    }
+    parallel(
+            findBugs: {
+                node {
+                    unstash name: 'built'
+                    // findBugs
+                    try {
+                        sh './gradlew -Dfindbugs.xml.report=true findbugsMain'
+                    } finally {
+                        step([$class: 'FindBugsPublisher', pattern: '**/build/reports/findbugs/*.xml'])
+                    }
+                }
+            },
+            cloudant: {
+                node {
+                    unstash name: 'built'
+                    // tests
+                    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'clientlibs-test', usernameVariable: 'DB_USER', passwordVariable: 'DB_PASSWORD']]) {
+                        try {
+                            sh './gradlew -Dtest.with.specified.couch=true -Dtest.couch.username=$DB_USER -Dtest.couch.password=$DB_PASSWORD -Dtest.couch.host=clientlibs-test.cloudant.com -Dtest.couch.port=443 -Dtest.couch.http=https -Dtest.couch.ignore.compaction=true -Dtest.couch.auth.headers=true cloudantServiceTest'
+                        } finally {
+                            junit '**/build/test-results/*.xml'
+                        }
+                    }
+                }
+            },
+            couchdb1dot6: {
+                node("couchdb1.6") {
```

Comment:
Although this uses DinD (Docker in Docker) to run the server we are testing against it appears to not work with the standard slave image, however it works with the image created to run CouchDB from our Artifactory. We can change this for DinD tag instead but i thought i'd get some feedback before making the changes.